planning/behavior_velocity_dynamic_obstacle_stop_module/src/object_filtering.cpp

```diff
@@ -59,13 +59,36 @@ std::vector<autoware_auto_perception_msgs::msg::PredictedObject> filter_predicte
     return obj_arc_length > ego_data.longitudinal_offset_to_first_path_idx +
                               params.ego_longitudinal_offset + o.shape.dimensions.x / 2.0;
   };
-  for (const auto & object : objects.objects)
+  const auto is_unavoidable = [&](const auto & o) {
+    const auto & o_pose = o.kinematics.initial_pose_with_covariance.pose;
+    const auto o_yaw = tf2::getYaw(o_pose.orientation);
+    const auto ego_yaw = tf2::getYaw(ego_data.pose.orientation);
+    const auto yaw_diff = std::abs(tier4_autoware_utils::normalizeRadian(o_yaw - ego_yaw));
+    const auto opposite_heading = yaw_diff > M_PI_2 + M_PI_4;
+    const auto collision_distance_threshold =
+      params.ego_lateral_offset + o.shape.dimensions.y / 2.0 + params.hysteresis;
+    auto has_collision = std::abs(
+                           (o_pose.position.y - ego_data.pose.position.y) * std::cos(o_yaw) -
+                           (o_pose.position.x - ego_data.pose.position.x) * std::sin(o_yaw)) <=
```

Comment:
What sort of distance is this line calculating ?

Reply:
This is the lateral distance of the ego vehicle, relative to the object. This is taken from the following: https://en.wikipedia.org/wiki/Distance_from_a_point_to_a_line#Line_defined_by_point_and_angle
I made it more clear: [de8fc13](https://github.com/autowarefoundation/autoware.universe/pull/6594/commits/de8fc1305dc7d99b22dbd132e6874892c1b6af62)